pkg/khstatecrd/functions.go

```diff
@@ -101,15 +100,19 @@ func (c *KuberhealthyStateClient) List(opts metav1.ListOptions, resource string,
 }
 
 // Watch returns a watch.Interface that watches the requested clusterTestTypes.
-func (c *KuberhealthyStateClient) Watch(opts metav1.ListOptions) (watch.Interface, error) {
+func (c *KuberhealthyStateClient) Watch(opts metav1.ListOptions, resource string, namespace string) (watch.Interface, error) {
 	var timeout time.Duration
 	if opts.TimeoutSeconds != nil {
 		timeout = time.Duration(*opts.TimeoutSeconds) * time.Second
 	}
 	opts.Watch = true
-	return c.restClient.Get().
-		Resource("khstates").
+
+	clientResponse, err := c.restClient.Get().
+		Resource(resource).
+		Namespace(namespace).
 		VersionedParams(&opts, scheme.ParameterCodec).
 		Timeout(timeout).
 		Watch()
+
+	return clientResponse, err
```

Comment:
You can change this to simply return:

```suggestion
	return c.restClient.Get().
		Resource(resource).
		Namespace(namespace).
		VersionedParams(&opts, scheme.ParameterCodec).
		Timeout(timeout).
		Watch()
```